### PR TITLE
Release v0.13.0-rc1

### PR DESCRIPTION
**Pull request by johnugeorge**
Created release as per instructions in https://github.com/kserve/kserve/blob/master/release/RELEASE_PROCESS_v2.md

Note: https://github.com/kserve/kserve/pull/3700 is also cherrypicked to release branch

### CODE REVIEW

**Review by terrytangyuan — APPROVED** [review]
/lgtm

File: python/xgbserver/poetry.lock [hunk 1]
```diff
@@ -1363,7 +1363,7 @@ referencing = ">=0.31.0"
 
 [[package]]
 name = "kserve"
-version = "0.12.1"
+version = "0.13.0rc0"
```
Comment (sivanantha321) [inline python/xgbserver/poetry.lock]: This should be 0.13.0rc1

File: python/paddleserver/poetry.lock [hunk 2]
```diff
@@ -1374,7 +1374,7 @@ referencing = ">=0.31.0"
 
 [[package]]
 name = "kserve"
-version = "0.12.1"
+version = "0.13.0rc0"
```
Comment (sivanantha321) [inline python/paddleserver/poetry.lock]: Should be 0.13.0rc1

File: python/lgbserver/poetry.lock [hunk 3]
```diff
@@ -1363,7 +1363,7 @@ referencing = ">=0.31.0"
 
 [[package]]
 name = "kserve"
-version = "0.12.1"
+version = "0.13.0rc0"
```
Comment (sivanantha321) [inline python/lgbserver/poetry.lock]: should be 0.13.0rc1

File: python/aiffairness/poetry.lock [hunk 4]
```diff
@@ -982,7 +982,7 @@ referencing = ">=0.31.0"
 
 [[package]]
 name = "kserve"
-version = "0.12.1"
+version = "0.13.0rc0"
```
Comment (sivanantha321) [inline python/aiffairness/poetry.lock]: Should be 0.13.0rc1

File: python/xgbserver/poetry.lock (hunk 1 above)
Comment (yuzisun) [reply python/xgbserver/poetry.lock]: @sivanantha321 any idea why CI did not complain ?

**Comment by johnugeorge** [comment]
Pushed the changes.

File: python/xgbserver/poetry.lock (hunk 1 above)
Comment (sivanantha321) [reply python/xgbserver/poetry.lock]: #3707

**Pull request merged**